inst/include/RcppArmadilloAs.h

```diff
@@ -403,14 +403,61 @@ namespace traits {
                     res = symmatl(res);
                 }
             }
+            else if (type == "pMatrix") {
+                IntegerVector i(ncol);
+                IntegerVector p(ncol+1);
```

Comment:
This points to a big problem I never thought of. 
When 
1. nrow = ncol;
2. all the non-zero values are the same;
3. each row and each column only has a non-zero values

The converted dgCMatrix will not carry all the information of the matrix, which means the further mathematic operation based on the converted dgCMatrix might be meaningless.